test/new-e2e/tests/cws/e2e_cws_test.go

```diff
@@ -156,3 +154,17 @@ func (a *agentSuite) TestOpenSignal() {
 	assert.Contains(a.T(), agentContext["rule_id"], a.agentRuleName, "unable to find tag")
 
 }
+
+func (a *agentSuite) waitAgentLogs(agentName string, pattern string) error {
+	err := backoff.Retry(func() error {
+		output, err := a.Env().VM.ExecuteWithError(fmt.Sprintf("cat /var/log/datadog/%s.log", agentName))
+		if err != nil {
+			return err
+		}
+		if strings.Contains(output, pattern) {
+			return nil
+		}
+		return errors.New("no log found")
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(500*time.Millisecond), 60))
+	return err
+}
```

Comment:
I’m not familiar enough with the VM API to make a concrete counter proposition but there are two things that look slightly sub-optimal to me:
* We are loading the the content of the full file in memory in one time. If this function is called after a while and the agent has enabled debug logs, it can trigger a big memory consumption.
* Reading the whole file again and again is suboptimal. It would be more efficient to have something like a `tail -f`.

Reply:
Honestly I agree with you that all of this is far from perfect. My goal with this PR is to take what was built by Momar and to have it passing the CI. Further improvements will follow, without the need to ping 3 teams in the process

Reply:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You could use testify `Eventually` instead of backoff, it provides retry logic - can be addressed in a follow up.

You could also test against fakeintake or DD intake for a better blackbox test.

I like @L3n41c suggestion, you could have a `tail -f` with a filter that stores only interesting lines, and can send an event when the filtered lines are read. Agreed can be part of a follow up